app/Http/Controllers/CourseController.php

```diff
@@ -14,7 +14,16 @@ class CourseController extends Controller
      */
     public function index()
     {
-        $courses = Course::all();
+        $user = request()->user()->load('student');
+
+        $courses = [];
+        if ($user->isAdmin()) {
+            $courses = Course::all();
+        }
+
+        if ($user->student->course_id) {
```

Comment:
Esto me está fallando cuando entro como admin y no tengo un `Student` asociado a mi `User`.

```suggestion
        if ($user->student && $user->student->course_id) {
```